### PR TITLE
Ddd

### PR DESCRIPTION
**Pull request by VolodymyrM27**
@DmitryOnopriienko

### CODE REVIEW

File: airguardian-core/battery-application/src/main/kotlin/systems/ajax/motrechko/airguardian/batteryapplication/application/port/DroneChargingApplicationProducerOutPort.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package systems.ajax.motrechko.airguardian.batteryapplication.application.port
+
+import reactor.core.publisher.Mono
+import systems.ajax.motrechko.airguardian.commonresponse.application.drone_battery_charging_application.proto.DroneBatteryChargingApplication
```
Comment (ajax-gnatiuk-s) [inline airguardian-core/battery-application/src/main/kotlin/systems/ajax/motrechko/airguardian/batteryapplication/application/port/DroneChargingApplicationProducerOutPort.kt:4]: `DroneBatteryChargingApplication` is DTO that is part of infrastructure layer. Application layer should not depend on infrastructure layer

File: airguardian-core/drone/src/main/kotlin/systems/ajax/motrechko/airguardian/drone/application/util/BatteryCalculator.kt [hunk 2]
```diff
@@ -1,6 +1,6 @@
-package systems.ajax.motrechko.airguardian.utils
+package systems.ajax.motrechko.airguardian.drone.application.util
```
Comment (ajax-gnatiuk-s) [inline airguardian-core/drone/src/main/kotlin/systems/ajax/motrechko/airguardian/drone/application/util/BatteryCalculator.kt:1]: Now this class looks more like a service. It's not call to action, just informing you that it's more correct making the class `@Service`

File: airguardian-core/drone/src/main/kotlin/systems/ajax/motrechko/airguardian/drone/infrastructure/adapters/repository/entity/MongoDrone.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.data.mongodb.core.mapping.Document
+import systems.ajax.motrechko.airguardian.drone.domain.Drone
+import systems.ajax.motrechko.airguardian.drone.domain.DroneSize
+import systems.ajax.motrechko.airguardian.drone.domain.DroneStatus
+import systems.ajax.motrechko.airguardian.drone.domain.DroneType
+
+@Document("drone")
+data class MongoDrone(
+    @Id
+    @JsonSerialize(using = ToStringSerializer::class)
```
Comment (ajax-gnatiuk-s) [inline airguardian-core/drone/src/main/kotlin/systems/ajax/motrechko/airguardian/drone/infrastructure/adapters/repository/entity/MongoDrone.kt:16]: do we still need the annotation?

File: airguardian-core/drone/src/main/kotlin/systems/ajax/motrechko/airguardian/drone/infrastructure/adapters/repository/redis/RedisDroneRepository.kt [hunk 4]
```diff
@@ -52,18 +64,23 @@ class DroneRedisRepository(
...
     private fun saveDroneToRedisCache(drone: Drone): Mono<Drone> {
-        val key = drone.id.toHexString()
-        return reactiveRedisRepository.opsForValue()
+        val key = drone.id
+        return if (key != null) {
+            reactiveRedisRepository.opsForValue()
                 .set(
                     dronePrefix + key,
-                    drone,
+                    drone.toMongoDrone(),
```
Comment (ajax-gnatiuk-s) [inline airguardian-core/drone/src/main/kotlin/systems/ajax/motrechko/airguardian/drone/infrastructure/adapters/repository/redis/RedisDroneRepository.kt:79]: Storing a drone in redis: 
  `Drone` (domain) -> MongoDrone -> store

Find in redis:
 storage -> MongoDrone -> Drone (domain) 

Looks like it should be reworked :)

File: airguardian-core/drone/src/main/kotlin/systems/ajax/motrechko/airguardian/drone/application/util/BatteryCalculator.kt (hunk 2 above)
Comment (VolodymyrM27) [reply airguardian-core/drone/src/main/kotlin/systems/ajax/motrechko/airguardian/drone/application/util/BatteryCalculator.kt:1]: Do I need to annotate `@Service` or rewrite this class and move it to the application layer?

File: airguardian-core/monitoring-object/src/main/kotlin/systems/ajax/motrechko/airguardian/monitoringobject/application/port/MonitoringObjectServiceInPort.kt [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import systems.ajax.motrechko.airguardian.monitoringobject.domain.MonitoringObject
+
+interface MonitoringObjectServiceInPort {
+    fun findAll(): Flux<MonitoringObject>
+
+    fun findById(id: String): Mono<MonitoringObject>
+
+    fun deleteById(id: String): Mono<Unit>
+    fun save(monitoringObject: MonitoringObject): Mono<MonitoringObject>
+    fun update(monitoringObject: MonitoringObject): Mono<MonitoringObject>
```
Comment (ajax-gnatiuk-s) [inline airguardian-core/monitoring-object/src/main/kotlin/systems/ajax/motrechko/airguardian/monitoringobject/application/port/MonitoringObjectServiceInPort.kt:14]:
```suggestion

    fun findAll(): Flux<MonitoringObject>

    fun findById(id: String): Mono<MonitoringObject>

    fun deleteById(id: String): Mono<Unit>
    
    fun save(monitoringObject: MonitoringObject): Mono<MonitoringObject>
    
    fun update(monitoringObject: MonitoringObject): Mono<MonitoringObject>
```

**Review by ajax-gnatiuk-s — CHANGES_REQUESTED** [review]

**Review by ajax-gnatiuk-s — APPROVED** [review]

**Pull request merged**